### DISCUSSION TITLE
setCompany not working?

### DISCUSSION THREAD

**Discussion by neildempster** [PHPOffice/PhpSpreadsheet#3083] [Q&A] [answered]
I am trying to set the worksheet "Properties" when creating a spreadsheet with PHPSpreadsheet. All fields are working with the code below EXCEPT for setCompany. When I look at the resulting worksheet properties, that field states "Specify the company" so, clearly, PHPSpreadsheet is not updating that field even though all the others are updating properly. Any ideas on how to fix this?

```
$spreadsheet->getProperties()
            ->setCreator("CPS Report Generator") // Excel Properties: "Author"
            ->setLastModifiedBy("CPS Report Generator") // Excel Properties: "Last Modified By"
            ->setCompany("CPS Inc.")
            ->setTitle($prop_title) // Excel Properties: "Title"
            ->setSubject("CPS Report Generator Export Facility") // Excel Extended Properties: "Subject"
            ->setDescription("Report from CPS") // Excel Extended Properties: "Comments"
            ->setKeywords($prop_tags) // Excel Properties: "Tags"
            ->setCategory($prop_categories); // Excel Properties: "Categories"

```
In the PHPSpreadsheet help docs, it lists setCompany as an option so I am stymied.

**Comment by MarkBaker** [comment, marked as answer]
What format are you writing? Company is an extended property, and that isn't supported by the Xls Writer; but it is written correctly for Xlsx, and as a meta property for Ods

**Comment by neildempster** [comment]
I am generating an xls file. However, "Comments" (PHPSpreadsheet = setDescription) and "Subject" (PHPSpreadsheet = Subject) are also extended properties and they work? (at least how my Excel works under the "Show All Properties" that opens to show the extended.)

**Reply by MarkBaker** [reply]
https://github.com/PHPOffice/PhpSpreadsheet/blob/master/docs/references/features-cross-reference.md

The original list of basic properties comprised:
 - Creator
 - Creation Date/Time
 - Modifier
 - Modified Date/Time
 - Title
 - Description
 - Subject
 - Keywords

Extended properties were:
 - Category
 - Company
 - Manager

See https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-oshared/c0e0190a-0828-4638-ab74-6fc190da3582 for the relevant section of MS documentation

**Reply by neildempster** [reply]
THANK YOU! I was assuming that the 'hidden' properties in Excel were the 'extended' properties. I appreciate your input and the additional resource. Question is now answered (with my thanks).
